src/orders/components/OrderReturnPage/form.tsx

```diff
@@ -0,0 +1,215 @@
+    getAllOrderFulfilledLines(order).map(getParsedLineData(0, true))
+  );
+
+  const getItemsToBeReplaced = () => {
+    if (!order) {
+      return [];
+    }
+
+    const orderLinesItems = getOrderUnfulfilledLines(order).map(
+      getParsedLineData(true)
+    );
+
+    const fulfilmentsItems = getFulfilledFulfillemnts(order).reduce(
+      (result, { lines }) => [
+        ...result,
+        ...getParsedFulfiledLines(lines).map(getParsedLineData(true, true))
+      ],
+      []
+    );
+
+    return [...orderLinesItems, ...fulfilmentsItems];
+  };
+
+  const itemsToBeReplaced = useFormset<LineItemData, boolean>(
+    getItemsToBeReplaced()
+  );
+
+  const handleSetMaximalUnfulfiledItemsQuantities = () => {
+    const newQuantities: FormsetQuantityData = unfulfiledItemsQuantites.data.map(
+      ({ id }) => {
+        const line = order.lines.find(getById(id));
+        const value = line.quantity - line.quantityFulfilled;
+
+        return getLineItem(line, value);
+      }
+    );
+
+    triggerChange();
+    unfulfiledItemsQuantites.set(newQuantities);
+  };
+
+  const handleSetMaximalFulfiledItemsQuantities = (
+    fulfillmentId: string
+  ) => () => {
+    const { lines } = order.fulfillments.find(getById(fulfillmentId));
+
+    const newQuantities: FormsetQuantityData = fulfiledItemsQuatities.data.map(
+      item => {
+        const line = lines.find(getById(item.id));
+
+        if (!line) {
+          return item;
+        }
+
+        return getLineItem(line, line.quantity);
+      }
+    );
+
+    triggerChange();
+    fulfiledItemsQuatities.set(newQuantities);
+  };
+
+  const data: OrderReturnFormData = {
+    fulfiledItemsQuantities: fulfiledItemsQuatities.data,
+    itemsToBeReplaced: itemsToBeReplaced.data,
+    unfulfiledItemsQuantities: unfulfiledItemsQuantites.data,
+    ...form.data
+  };
+
+  const submit = () => handleFormSubmit(data, onSubmit, setHasChanged);
+
+  const triggerChange = () => setHasChanged(true);
+
+  function handleHandlerChange<T>(callback: (id: string, value: T) => void) {
+    return (id: string, value: T) => {
+      triggerChange();
+      callback(id, value);
+    };
+  }
+
+  return {
+    change: handleChange,
+    data,
+    handlers: {
+      changeFulfiledItemsQuantity: handleHandlerChange(
+        fulfiledItemsQuatities.change
+      ),
+      changeItemsToBeReplaced: handleHandlerChange(itemsToBeReplaced.change),
+      changeUnfulfiledItemsQuantity: handleHandlerChange(
+        unfulfiledItemsQuantites.change
+      ),
+      handleSetMaximalFulfiledItemsQuantities,
+      handleSetMaximalUnfulfiledItemsQuantities
+    },
+    hasChanged,
+    submit
+  };
+}
+
+const OrderReturnForm: React.FC<OrderReturnProps> = ({
+  children,
+  order,
+  onSubmit
+}) => {
+  const props = useOrderReturnForm(order, onSubmit);
+
+  return <form onSubmit={props.submit}>{children(props)}</form>;
+};
+
+OrderReturnForm.displayName = "OrderReturnForm";
```

Comment:
This shouldn't be needed, React takes the component name by default.

Reply:
That's right. Didn't actually check that, thought we were doing it because of...reasons? Created a [ticket](https://app.clickup.com/t/2549495/SALEOR-1922) to get that cleaned up

Reply:
These are useful when React cannot resolve the names for debugging (like Context names), you're free to remove it here though